### DISCUSSION TITLE
Swap groups

### DISCUSSION THREAD

**Discussion by gruvw** [qtile/qtile#4232] [Q&A]
Hey !

I am looking for a way two write a `swap_group` function that will simply swap every windows between two groups.

They keys I would like are the keychord `mod + g` and then simultaneously pressing the two group keys should swap the groups.
So `mod + g, 12` should swap group 1 with group 2.

I tried the following:

```python
# config.py
...

groups = [Group(i) for i in "123456789"]

...

@lazy.function
def swap_groups(qtile, i, gi, j, gj):
    # is this correct ?
    gi.name, gj.name = gj.name, gi.name
    groups[i], groups[j] = groups[j], groups[i]

...

keys += [
    KeyChord([mod], "g", [
        # Probably an issue here...
        Key([gi.name], gj.name, swap_groups(i, gi, j, gj),
            desc=f"Swaps windows between {gi.name} and {gj.name}")
        for (i, gi) in enumerate(groups) for (j, gj) in enumerate(groups) if gi.name != gj.name
    ], name="group"),
]

...
```

But this is not working out... As soon as I press `mod + g` I totally break the keys and nothing works anymore, even pressing `esc` does not get me out of the `group` keychord.

Could you please help me write the config for the intended behavior ?

Thank you so much !

**Comment by tych0** [comment]
Does `lazy.group.toscreeen(toggle=True)` help you here?

**Reply by gruvw** [reply]
Well not really... I don't want to go to that group, I just want to swap the two groups.

If I have windows W1, W2 on group 1 and window W3 on group 2, I would like to have W3 on group 1 and W1, W2 on group 2.

**Reply by tych0** [reply]
Ah, you want to swap the windows in the group instead of the groups themselves? Your code here doesn't show what the values of gi/gj are bound to, but you'll need to iterate over the windows for both and insert them into the other groups.

**Reply by gruvw** [reply]
> Your code here doesn't show what the values of gi/gj are bound to

`for (i, gi) in enumerate(groups) for (j, gj) in enumerate(groups) if gi.name != gj.name`

> iterate over the windows for both and insert them into the other groups

Won't that mess up their order and placement ?

**Comment by tych0** [comment]
> Won't that mess up their order and placement ?

yes, you'll need to do it carefully to get the ordering right. it may require patches to qtile.

**Reply by gruvw** [reply]
Could you provide the code for the `swap_groups` function please ?
